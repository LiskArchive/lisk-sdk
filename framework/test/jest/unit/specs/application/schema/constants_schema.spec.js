/*
 * Copyright Â© 2019 Lisk Foundation
 *
 * See the LICENSE file at the top-level directory of this distribution
 * for licensing information.
 *
 * Unless otherwise agreed in a custom licensing agreement with the Lisk Foundation,
 * no part of this software, including this file, may be copied, modified,
 * propagated, or distributed except according to the terms contained in the
 * LICENSE file.
 *
 * Removal or modification of this copyright notice is prohibited.
 */

'use strict';

const constantsSchema = require('../../../../../../src/application/schema/constants_schema');

describe('schema/constants_schema.js', () => {
	it('constants schema must match to the snapshot.', () => {
		expect(constantsSchema).toMatchSnapshot();
	});
});
