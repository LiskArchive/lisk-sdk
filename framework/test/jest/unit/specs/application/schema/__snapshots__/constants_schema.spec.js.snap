// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`schema/constants_schema.js constants schema must match to the snapshot. 1`] = `
Object {
  "additionalProperties": false,
  "default": Object {
    "ACTIVE_DELEGATES": 101,
    "BLOCK_RECEIPT_TIMEOUT": 20,
    "BLOCK_SLOT_WINDOW": 5,
    "EXPIRY_INTERVAL": 30000,
    "FEES": Object {
      "DAPP_DEPOSIT": "10000000",
      "DAPP_REGISTRATION": "2500000000",
      "DAPP_WITHDRAWAL": "10000000",
      "DELEGATE": "2500000000",
      "MULTISIGNATURE": "500000000",
      "SECOND_SIGNATURE": "500000000",
      "SEND": "10000000",
      "VOTE": "100000000",
    },
    "MAX_PAYLOAD_LENGTH": 1048576,
    "MAX_SHARED_TRANSACTIONS": 100,
    "MAX_VOTES_PER_ACCOUNT": 101,
    "TOTAL_AMOUNT": "10000000000000000",
    "TRANSACTION_TYPES": Object {
      "DAPP": 5,
      "DELEGATE": 2,
      "IN_TRANSFER": 6,
      "MULTI": 4,
      "OUT_TRANSFER": 7,
      "SEND": 0,
      "SIGNATURE": 1,
      "VOTE": 3,
    },
    "UNCONFIRMED_TRANSACTION_TIMEOUT": 10800,
  },
  "id": "#constants",
  "properties": Object {
    "ACTIVE_DELEGATES": Object {
      "const": 101,
      "description": "The default number of delegates allowed to forge a block",
      "format": "oddInteger",
      "min": 1,
      "type": "number",
    },
    "BLOCK_RECEIPT_TIMEOUT": Object {
      "const": 20,
      "description": "Seconds to check if the block is fresh or not",
      "min": 1,
      "type": "integer",
    },
    "BLOCK_SLOT_WINDOW": Object {
      "const": 5,
      "description": "The default number of previous blocks to keep in memory",
      "min": 1,
      "type": "integer",
    },
    "EXPIRY_INTERVAL": Object {
      "const": 30000,
      "description": "Transaction pool expiry timer in milliseconds",
      "min": 1,
      "type": "integer",
    },
    "FEES": Object {
      "additionalProperties": false,
      "description": "Object representing amount of fees for different types of transactions",
      "properties": Object {
        "DAPP_DEPOSIT": Object {
          "const": "10000000",
          "format": "amount",
          "type": "string",
        },
        "DAPP_REGISTRATION": Object {
          "const": "2500000000",
          "description": "Fee for registering as a dapp",
          "format": "amount",
          "type": "string",
        },
        "DAPP_WITHDRAWAL": Object {
          "const": "10000000",
          "format": "amount",
          "type": "string",
        },
        "DELEGATE": Object {
          "const": "2500000000",
          "description": "Fee for registering as a delegate",
          "format": "amount",
          "type": "string",
        },
        "MULTISIGNATURE": Object {
          "const": "500000000",
          "description": "Fee for multisignature transaction",
          "format": "amount",
          "type": "string",
        },
        "SECOND_SIGNATURE": Object {
          "const": "500000000",
          "description": "Fee for creating a second signature",
          "format": "amount",
          "type": "string",
        },
        "SEND": Object {
          "const": "10000000",
          "description": "Fee for sending a transaction",
          "format": "amount",
          "type": "string",
        },
        "VOTE": Object {
          "const": "100000000",
          "description": "Fee for voting a delegate",
          "format": "amount",
          "type": "string",
        },
      },
      "required": Array [
        "SEND",
        "VOTE",
        "SECOND_SIGNATURE",
        "DELEGATE",
        "MULTISIGNATURE",
        "DAPP_REGISTRATION",
        "DAPP_WITHDRAWAL",
        "DAPP_DEPOSIT",
      ],
      "type": "object",
    },
    "MAX_PAYLOAD_LENGTH": Object {
      "const": 1048576,
      "description": "Maximum transaction bytes length for 25 transactions in a single block",
      "min": 1,
      "type": "integer",
    },
    "MAX_SHARED_TRANSACTIONS": Object {
      "const": 100,
      "description": "Maximum number of in-memory transactions/signatures shared across peers",
      "min": 1,
      "type": "integer",
    },
    "MAX_VOTES_PER_ACCOUNT": Object {
      "const": 101,
      "description": "The maximum number of votes allowed in transaction type(3) votes",
      "maximum": Object {
        "$data": "/ACTIVE_DELEGATES",
      },
      "min": 1,
      "type": "number",
    },
    "TOTAL_AMOUNT": Object {
      "const": "10000000000000000",
      "description": "Total amount of LSK available in network before rewards milestone started",
      "format": "amount",
      "type": "string",
    },
    "TRANSACTION_TYPES": Object {
      "additionalProperties": false,
      "properties": Object {
        "DAPP": Object {
          "const": 5,
          "type": "integer",
        },
        "DELEGATE": Object {
          "const": 2,
          "type": "integer",
        },
        "IN_TRANSFER": Object {
          "const": 6,
          "type": "integer",
        },
        "MULTI": Object {
          "const": 4,
          "type": "integer",
        },
        "OUT_TRANSFER": Object {
          "const": 7,
          "type": "integer",
        },
        "SEND": Object {
          "const": 0,
          "type": "integer",
        },
        "SIGNATURE": Object {
          "const": 1,
          "type": "integer",
        },
        "VOTE": Object {
          "const": 3,
          "type": "integer",
        },
      },
      "required": Array [
        "SEND",
        "SIGNATURE",
        "DELEGATE",
        "VOTE",
        "MULTI",
        "DAPP",
        "IN_TRANSFER",
        "OUT_TRANSFER",
      ],
      "type": "object",
    },
    "UNCONFIRMED_TRANSACTION_TIMEOUT": Object {
      "const": 10800,
      "description": "Expiration time for unconfirmed transaction/signatures in transaction pool",
      "min": 1,
      "type": "integer",
    },
  },
  "required": Array [
    "ACTIVE_DELEGATES",
    "BLOCK_SLOT_WINDOW",
    "BLOCK_RECEIPT_TIMEOUT",
    "FEES",
    "MAX_PAYLOAD_LENGTH",
    "MAX_SHARED_TRANSACTIONS",
    "MAX_VOTES_PER_ACCOUNT",
    "TOTAL_AMOUNT",
    "TRANSACTION_TYPES",
    "UNCONFIRMED_TRANSACTION_TIMEOUT",
    "EXPIRY_INTERVAL",
  ],
  "type": "object",
}
`;
